### ISSUE TITLE
(apigateway): set latestDeployment on RestApi when using fromRestApiAttributes() to trigger API deployments 

### ISSUE THREAD

**Issue by tmclaugh** [aws/aws-cdk#13383]
We are adopting a model where the APIG RestApi lives in a stack and Resources and Methods live independently in other stacks in other repositories. In the Resources and Methods stacks we use [_RestApi.fromRestApiAttributes()_ to create the RestApi object.

This approach has two known issues:

* Need to properly set dependencies on the Deployment resource
* Need to modify the Deployment LogicalId on changes to API related resources.

I want this to work seamlessly without having to search for resources in the stack for dependencies, know which ones to exclude (I see you AWS::Lambda::Permission 👈👀), and feed data from all those dependencies to _Deployment.addToLogicalId()_ so updates trigger a deploy by CloudFormation.

### Proposed Solution

Setting __latestDeployment_ when calling _RestApi.fromRestApiAttributes()_ appears to properly solve my issue but I'm not yet aware repercussions to doing this.

### Other

* [X] :wave: I may be able to implement this feature request
* [X] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by nija-at** [comment]
The consequences of setting latest deployment is that, the deployment cannot be automatically updated when your resources or methods are updated. This is misleading as the CDK makes you assume that the 'latest deployment' will have all the updates as of the last 'cdk deploy'.

We're tracking this here - https://github.com/aws/aws-cdk/issues/12417

Until then, the choice is that the user configures the deployment themselves and updates the deployment when necessary.

**Comment by tmclaugh** [comment]
> the deployment cannot be automatically updated when your resources or methods are updated

That doesn't reflect the results of my own experimentation. By setting _latestDeployment_ by creating a nw Deployment resource inside of _fromRestApiAttributes()_ I triggered updates on the Deployment LogicalId on each change to a Resource or Method. All I did was crudely recreate the logic that occurs when a RestApi object is constructed. It's not perfect but it's a start at tackling this long standing issue.
